core/plugins/org.polarsys.capella.core.data.la.validation/src/org/polarsys/capella/core/data/la/validation/logicalComponent/LogicalSystem_RealizedSystemSystem.java

```diff
@@ -37,8 +38,17 @@ public IStatus validate(IValidationContext ctx) {
         LogicalComponent component = (LogicalComponent) eObj;
         if (component.equals(BlockArchitectureExt.getRootBlockArchitecture(component).getSystem())) {
           if (component.getRealizedSystemComponents().isEmpty()) {
-            return ctx.createFailureStatus(
-                CapellaElementExt.getValidationRuleMessagePrefix(component) + " does not realize the System Component");
+            String previousRootCompoenentname = "System Component";
```

Comment:
typo.
mainly, can you use getMetaclasslabel instead. 
the Root keyword seems duplicated. Root function is called "Root system function. it will appera twice.

code between several rules seems duplicated. can you make a class used in several rules?

Reply:
That's why I added " " for the name to differentiate it from the name.

Reply:
getMetaclassLabel(component) will return "System Component"

Reply:
component.getName will return "SystemComponent 1"

Reply:
previousRootCompoenentname  => previousRootComponentname